pkg/controller/sdk/status_accessor.go

```diff
@@ -44,10 +46,10 @@ func NewReflectedStatusAccessor(object interface{}) StatusAccessor {
 		if _, ok := statusField.Interface().(interface{}); ok {
 			return &reflectedStatusAccessor{
 				status: statusField,
-			}
+			}, nil
```

Comment:
Optional: I find code easier to read when the happy path is the final path. In this case, the happy path is the next to final path, and to understand that I need to understand that the condition is meant to be positive rather than negative.

The way the functions are named, it's weird for them to be a negative condition, so I understand why it was written this way. It's just not what I was expecting. :grin: